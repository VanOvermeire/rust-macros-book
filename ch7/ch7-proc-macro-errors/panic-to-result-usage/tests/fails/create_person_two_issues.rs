use panic_to_result_macro::panic_to_result;

#[derive(Debug)]
pub struct Person {
    name: String,
    age: u32,
}

#[panic_to_result]
fn create_person_two_issues(name: String, age: u32) -> Result<String, Person> {
    if age > 30 {
        panic!();
    }
    Ok(Person {
        name,
        age,
    })
}

fn main() {}
