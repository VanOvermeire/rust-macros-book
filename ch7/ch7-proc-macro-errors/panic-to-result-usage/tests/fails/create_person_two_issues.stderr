error: panic needs a message!

         = help: try to add a message: panic!("Example".to_string())
         = note: we will add the message to Result's Err

  --> tests/fails/create_person_two_issues.rs:12:9
   |
12 |         panic!();
   |         ^^^^^^^^^

error: this macro can only be applied to a function that does not yet return a Result. Signature: Result < String, Person >
  --> tests/fails/create_person_two_issues.rs:10:56
   |
10 | fn create_person_two_issues(name: String, age: u32) -> Result<String, Person> {
   |                                                        ^^^^^^^^^^^^^^^^^^^^^^
